kategory-effects/src/main/kotlin/kategory/effects/data/DeferredKW.kt

```diff
@@ -0,0 +1,110 @@
+package kategory
+
+import kategory.effects.data.internal.Platform
+import kategory.effects.instances.DeferredKWInstances
+import kotlinx.coroutines.experimental.*
+import kotlin.coroutines.experimental.CoroutineContext
+
+typealias DeferredResult<A> = Either<Throwable, A>
+
+@higherkind data class DeferredKW<out A>(val coroutineContext: CoroutineContext, val thunk: (CoroutineContext) -> Deferred<A>) : DeferredKWKind<A> {
```

Comment:
We should probably set the context default value to `CommonPool` or whatever `async/await` uses as default

Reply:
I agree, let me address that alongside other corutine issues on a different PR. They need to be fixed for Monad and Comonad too.